src/function/mtk.jl

```diff
@@ -0,0 +1,60 @@
+struct AutoModelingToolkit <: AbstractADType end
+
+function instantiate_function(f, x, ::AutoModelingToolkit, p, num_cons=0)
+
+    sys = ModelingToolkit.modelingtoolkitize(OptimizationProblem(f, x, p))
+    println(sys)
+    if f.grad === nothing
+        grad_oop, grad_iip = ModelingToolkit.generate_gradient(sys, expression=Val{false})
+        grad(J, u) = (grad_iip(J, u, p); J)
+    else
+        grad = f.grad
+    end
+
+    if f.hess === nothing
+        hess_oop, hess_iip = ModelingToolkit.generate_hessian(sys, expression=Val{false})
+        hess(J, u) = (hess_iip(J, u, p); J)
+    else
+        hess = f.hess
+    end
+
+    if f.hv === nothing
+        hv = function (H, θ, v, args...)
+            res = ArrayInterface.zeromatrix(θ)
+            hess(res, θ, args...)
+            H .= res * v
+        end
```

Comment:
we should just generate the Hv equation instead of using the Hessian. That would be a lot more efficient.